icu4c/source/i18n/tibetancal.cpp

```diff
@@ -0,0 +1,573 @@
+ }
+
+
+/**
+ * Override Calendar to handle leap months and leap days properly.
+ */
+void TibetanCalendar::add(UCalendarDateFields field, int32_t amount, UErrorCode& status) {
+    if (U_FAILURE(status)) {
+        return;
+    }
+    switch (field) {
+    case UCAL_MONTH:
+        {
+            int32_t month = internalGet(UCAL_MONTH);
+            int32_t is_leap_month = internalGet(UCAL_IS_LEAP_MONTH);
+            int32_t monthCount = toMonthCount(internalGet(UCAL_YEAR), month, is_leap_month);
+            int32_t julianDay = trueDate(internalGet(UCAL_DAY_OF_MONTH), monthCount + amount);
+            handleComputeFields(julianDay, status); 
+            break;
+        }
+    case UCAL_DAY_OF_MONTH:
+        {
+            int32_t julianDay = internalGet(UCAL_JULIAN_DAY);
+            int32_t newJulianDay = julianDay + amount;
+            handleComputeFields(newJulianDay, status);
+            break;
+        }
+    default:
+        Calendar::add(field, amount, status);
+        break;
+    }
+}
+
+
+/**
+ * Override Calendar to handle leap months and leap days properly.
+ */
+void TibetanCalendar::add(EDateFields field, int32_t amount, UErrorCode& status) {
+    add((UCalendarDateFields)field, amount, status);
+}
+
+
+/**
+ * Override Calendar to handle leap months and leap days properly.
+ */
+void TibetanCalendar::roll(UCalendarDateFields field, int32_t amount, UErrorCode& status) {
+    if(U_FAILURE(status)) {
+        return;
+    }
+    switch (field) {
+    case UCAL_MONTH:
+        {
+            int32_t month = internalGet(UCAL_MONTH);
+            int32_t is_leap_month = internalGet(UCAL_IS_LEAP_MONTH);
+            int32_t monthCount = toMonthCount(internalGet(UCAL_YEAR), month, is_leap_month);     
+            int32_t julianDay = trueDate(internalGet(UCAL_DAY_OF_MONTH), monthCount + amount);
+            handleComputeFields(julianDay, status); 
+            break;
+        }
+    case UCAL_DAY_OF_MONTH:
+        {
+            int32_t julianDay_m = internalGet(UCAL_JULIAN_DAY);
+            int32_t newJulianDay = julianDay_m + amount;
+            handleComputeFields(newJulianDay, status);
+            break;
+        }
+    default:
+        Calendar::add(field, amount, status);
+        break;
+    }
+}
+
+
+/**
+ * Override Calendar to handle leap months and leap days properly.
+ */
+void TibetanCalendar::roll(EDateFields field, int32_t amount, UErrorCode& status) {
+    roll((UCalendarDateFields)field, amount, status);
+}
+
+
+/**
+  * Subclasses may override this method to compute several fields
+  * specific to each calendar system.  These are:
+  *
+  * <ul><li>ERA
+  * <li>YEAR
+  * <li>MONTH
+  * <li>DAY_OF_MONTH
+  * <li>DAY_OF_YEAR
+  * <li>EXTENDED_YEAR</ul>
+  *
+  * <p>The GregorianCalendar implementation implements
+  * a calendar with the specified Julian/Gregorian cutover date.
+  * @internal
+  */
+void TibetanCalendar::handleComputeFields(int32_t julianDay, UErrorCode &status) {
```

Comment:
change to 
````
```
void TibetanCalendar::handleComputeFields(int32_t julianDay, UErrorCode& ) {
```
because status is unused variable